orca-core/src/main/java/com/netflix/spinnaker/orca/events/StageListenerAdapter.java

```diff
@@ -58,62 +48,19 @@ public void onApplicationEvent(ExecutionEvent event) {
   }
 
   private void onStageStarted(StageStarted event) {
-    PipelineExecution execution = retrieve(event);
-    List<StageExecution> stages = execution.getStages();
-    stages.stream()
-        .filter(it -> it.getId().equals(event.getStageId()))
-        .findFirst()
-        .ifPresent(stage -> delegate.beforeStage(persister, stage));
+    delegate.beforeStage(event.getStage());
   }
 
   private void onStageComplete(StageComplete event) {
-    PipelineExecution execution = retrieve(event);
-    List<StageExecution> stages = execution.getStages();
-    stages.stream()
-        .filter(it -> it.getId().equals(event.getStageId()))
-        .findFirst()
-        .ifPresent(stage -> delegate.afterStage(persister, stage));
+    delegate.afterStage(event.getStage());
   }
 
   private void onTaskStarted(TaskStarted event) {
-    PipelineExecution execution = retrieve(event);
-    List<StageExecution> stages = execution.getStages();
-    stages.stream()
-        .filter(it -> it.getId().equals(event.getStageId()))
-        .findFirst()
-        .ifPresent(
-            stage ->
-                delegate.beforeTask(
-                    persister,
-                    stage,
-                    stage.getTasks().stream()
-                        .filter(it -> it.getId().equals(event.getTaskId()))
-                        .findFirst()
-                        .get()));
+    delegate.beforeTask(event.getStage(), event.getTask());
   }
 
   private void onTaskComplete(TaskComplete event) {
-    PipelineExecution execution = retrieve(event);
-    List<StageExecution> stages = execution.getStages();
-    ExecutionStatus status = event.getStatus();
-    stages.stream()
-        .filter(it -> it.getId().equals(event.getStageId()))
-        .findFirst()
-        .ifPresent(
-            stage ->
-                delegate.afterTask(
-                    persister,
-                    stage,
-                    stage.getTasks().stream()
-                        .filter(it -> it.getId().equals(event.getTaskId()))
-                        .findFirst()
-                        .get(),
-                    status,
-                    // TODO: not sure if status.isSuccessful covers all the weird cases here
-                    status.isSuccessful()));
-  }
-
-  private PipelineExecution retrieve(ExecutionEvent event) {
-    return repository.retrieve(event.getExecutionType(), event.getExecutionId());
+    StageExecution stage = event.getStage();
+    delegate.afterTask(stage, event.getTask(), stage.getStatus(), stage.getStatus().isSuccessful());
```

Comment:
major 🤦‍♂️, this should be `task.getStatus()` and not `stage.getStatus()`

Reply:
Fixed in https://github.com/spinnaker/orca/pull/4105